srv/salt/state/miniconda.sls

```diff
@@ -1,26 +1,36 @@
 # File:/srv/salt/miniconda.sls
 
-download_miniconda_installer:
+check_conda_install:
+  cmd.run:
+    - name: conda --version
+    - runas: root
+    - shell: /bin/bash
+    - env:
+      - PATH: /root/miniconda3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin
+
+ensure_miniconda_installer:
   file.managed:
     - name: /home/centos/Miniconda3-latest-Linux-x86_64.sh
     - source: https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
     - skip_verify: True
-    - keep_source: True  # keep to avoid re-downloading.
+    - keep_source: True  # Keep to avoid re-downloading.
     - mode: 700
+    - onfail:
+      - cmd: check_conda_install
+
+install_miniconda:
+  cmd.run:
+    - name: 'sudo /home/centos/Miniconda3-latest-Linux-x86_64.sh -b -u'
+    - onfail:
+      - cmd: check_conda_install
 
-add_miniconda_to_bashrc:  # Add first, so that check in install_miniconda will work if conda is already installed.
+ensure_miniconda_path_in_bashrc:
   file.replace:
     - name: ~/.bashrc
     - pattern: /^export PATH="/root/miniconda3/bin:$PATH"/
     - repl: export PATH="/root/miniconda3/bin:$PATH"
     - append_if_not_found: True
```

Comment:
still sexy, good job

Reply:
Cheers. Apparently it was unnecessary, but like a lot of this, I've picked up some useful options in Salt, and flexed/developed some bash/*nix muscles :)

Reply:
still useful example state, I actually went and found this code to demonstrate to a client yesterday

Reply:
:)

> I find the miniconda.sls file needs renamed descriptors. you currently have 'miniconda_installed,' 'miniconda_installer_present,' 'install_miniconda.' These descriptors are almost like a tongue tie, I'd recommend refactoring to something like: "check_for_miniconda, download_miniconda, install_miniconda." Even though we are reusing miniconda over and over in the names, the rest of the descriptor is an action verb of that states purpose.

Is there some kinda of convention I'm misunderstanding? I understand the similar names issue, but I presume, with the paradigm of 'say what you want, not how to get there' that the states should reflect that tense. So, you "require miniconda is installed" rather than "require a check that miniconda is installed".